gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/MRCompactor.java

```diff
@@ -46,16 +50,46 @@
 import gobblin.util.DatasetFilterUtils;
 import gobblin.util.HadoopUtils;
 
+import static gobblin.compaction.mapreduce.MRCompactor.FolderState.*;
+import static gobblin.compaction.mapreduce.MRCompactorJobRunner.Status.*;
+
 
 /**
  * MapReduce-based {@link gobblin.compaction.Compactor}.
  *
  * @author ziliu
  */
+
 public class MRCompactor implements Compactor {
 
   private static final Logger LOG = LoggerFactory.getLogger(MRCompactor.class);
-  private static final List<Job> RUNNING_MR_JOBS = Lists.newCopyOnWriteArrayList();
+
+  private static final double HIGH_PRIORITY = 3.0;
+  private static final double NORMAL_PRIORITY = 2.0;
+  private static final double LOW_PRIORITY = 1.0;
+  private static final double PRIORITY_REDUCTION_FACTOR = 1.0 / 3.0;
+  private static final long COMPACTION_JOB_WAIT_INTERVAL_SECONDS = 10;
+  private static final Map<String, Job> RUNNING_MR_JOBS = Maps.newConcurrentMap();
+
+  public enum FolderState {
+
+    // An input folder with status NO indicates that the verification of its data completeness has not started.
+    // If data completeness verification is enabled, compaction should not start on an input folder if its status is
+    // NO, since we won't know whether its data is complete or not.
+    NO,
+
+    // PENDING indicates that the data completeness verification of a folder has started but not yet finished.
+    // Compaction may start on a folder with PENDING status. If, when the compaction job finishes, its status is still
+    // PENDING, the job won't publish data. If during the compaction job, the data verification finishes and the status
+    // changes to NO, the compaction job will be killed.
+    PENDING,
+
+    // YES indicates that data of an input folder is complete.
+    YES,
```

Comment:
I liked YES and NO from aesthetical point of view but it's fine if you prefer VERIFICATION_PASSED